File: lab2/fibo+memorization.cpp
// fibonacci series with memorization
//fibonacci series = 1,1,2,3,5,8,13...



#include <iostream>
using namespace std;


#define MAX 10


int table[MAX] = {0};

int fiboM(int n)
{
    if (n == 1 || n == 2)
    {
        return 1;
    }
    else
    {
        if (table[n] != 0)                                        //check if table[n] is non zero. 
        {
            return table[n];
        }
        else{
            table[n] = fiboM(n - 1) + fiboM(n - 2);
            return table[n];

        }
     
    }
}


int main()
{

    cout << fiboM(5);
}

File: lab2/fibo+memorization.py

MAX = 10
table = [0] * MAX      # a table of size 10 with all elements 0
 
def fiboM(n):
    if n == 1 or n==2:
        return 1
    else:
       if table[n] != 0:
           return table[n]
       else:
           table[n] = fiboM(n-1) + fiboM(n-2)
           return table[n]
       
print (fiboM(5))
    
